### DISCUSSION TITLE
New Diagram Style:  Sankey diagrams

### DISCUSSION THREAD

**Discussion by contentfree** [mermaid-js/mermaid#4474] [Ideas]
[Sankey diagrams](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fi.stack.imgur.com%2FcYsNT.png&f=1&nofb=1&ipt=925f51d08811430d7e2e5c09d87c411aebd508c65282e3f5cc6d8dbdc4d1bdda&ipo=images) are great for showing flows between different elements. 

For example, see https://i.stack.imgur.com/cYsNT.png 

Seems like a wonderful candidate for Mermaid and I hope you'll consider it.

**Comment by nirname** [comment]
I would gladly implement these, I was looking to that direction for quite a time. Having no experience with d3 directly, I have had some with parsers. So I'd like to hear syntax proposals from the community before I dive in.

However author of d3 already provide us with something, which is usually a csv, like this one:

```
Nuclear,Thermal,50
Gas,Thermal,10,orange
Solid,Thermal,30,green
Thermal,Losses,60
Thermal,Electricity,30
```

These values are not converted to percentage, these are just values. Link colors are not limited to simply colors, they can have some transition, as well as attributes of links are not limited to colors only. Apart from this, there are some graph-specified attributes links and nodes sort, margins, alignments, etc.

Currently I am researching on that topic

**Reply by nirname** [reply]
I also would like to use Docker for local development https://github.com/orgs/mermaid-js/discussions/4479

**Comment by nirname** [comment]
I've opened a [new issue](https://github.com/mermaid-js/mermaid/issues/4492) for that

**Comment by nirname** [comment]
I don't know if there is a way to mark discussion as resolved, but I have implemented first Sankey diagram prototype

**Reply by Yokozuna59** [reply]
You can close the discussion; if the category was `Q&A` then you could mark the comment as `answered`, but it's not, so closing it would be enough.

**Comment by Yokozuna59** [comment]
It has been implemented in #4502 by @nirname and merged, you'll be able to use it in the next release.
